samza-core/src/main/java/org/apache/samza/runtime/LocalApplicationRunner.java

```diff
@@ -247,7 +247,6 @@ StreamProcessor createStreamProcessor(
     latch.await();
 
     if (throwable.get() != null) {
-      appStatus = ApplicationStatus.UnsuccessfulFinish;
       throw throwable.get();
```

Comment:
I assume appStatus is no longer set here because the next line throws an exception. But on line 136, appStatus is set under the same circumstances. Why is that? Shouldn't run() and waitForFinish() have the same semantics for exception and status handling?